bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -0,0 +1,221 @@
+#ifndef _WIN32
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/utsname.h>
+#include <unistd.h>
+#include <gnu/libc-version.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <curl/curl.h>
+#include "old_machine_common_functions.h"
+
+#define BOOTLOADER_SERVER_PORT ":5001"
+
+struct response {
+  char *ptr;
+  size_t len;
+};
+
+void init_response(struct response *s) {
+  s->len = 0;
+  s->ptr = malloc(s->len+1);
+  if (s->ptr == NULL)
+    error("malloc() failed\n");
+  s->ptr[0] = '\0';
+}
+
+size_t writefunc(void *ptr, size_t size, size_t nmemb, struct response *s)
+{
+  size_t new_len = s->len + size*nmemb;
+  s->ptr = realloc(s->ptr, new_len+1);
+  if (s->ptr == NULL)
+    error("realloc() failed\n");
+
+  memcpy(s->ptr+s->len, ptr, size*nmemb);
+  s->ptr[new_len] = '\0';
+  s->len = new_len;
+
+  return size*nmemb;
+}
+
+char* executeCommand(char* commandLine) {
+    FILE *fp;
+    const int maxOutputLength = 2400;
+    char* fullOutput = (char *) malloc(maxOutputLength);
+    if(fullOutput == NULL){
+        error("Memory allocation failed\n");
+    }
+    /* Open the command for reading. */
+    fp = popen(commandLine, "r");
+    if (fp == NULL)
+        return ("Failed to run command\n" );
+
+    /* Read the output a line at a time - output it. */
+    char* res = fgets(fullOutput, maxOutputLength, fp);
+    if (res == NULL)
+        return("ERROR reading commandline\n");
+
+    /* close */
+    pclose(fp);
+
+    // Reallocate less memory
+    realloc(fullOutput, sizeof(fullOutput));
```

Comment:
sizeof(fullOutput) is sizeof(char*), always trimming the result. You want strlen.

sizeof is a compile time operator that returns the size of the variable, not the content.